bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniDrugOrderServiceImpl.java

```diff
@@ -167,6 +179,103 @@ public List<Order> getAllDrugOrders(String patientUuid, String patientProgramUui
+            Map<String, DrugOrder> drugOrderMap = getDiscontinuedDrugOrders(drugOrderList);
+            Collection<BahmniObservation> orderAttributeObs = bahmniObsService.observationsFor(patientUuid, getOrderAttributeConcepts(), null, null, false, null, null, null);
+            List<BahmniDrugOrder> bahmniDrugOrderList = bahmniDrugOrderMapper.mapToResponse(drugOrderList, orderAttributeObs, drugOrderMap , null);
+            Collections.sort(bahmniDrugOrderList, new Comparator<BahmniDrugOrder>() {
+                @Override
+                public int compare(BahmniDrugOrder o1, BahmniDrugOrder o2) {
+                    return o1.getEffectiveStartDate().compareTo(o2.getEffectiveStartDate());
+                }
+            });
+            return bahmniDrugOrderList;
+        } catch (IOException e) {
+            logger.error("Could not parse drug order", e);
+            throw new RuntimeException("Could not parse drug order", e);
+        }
+    }
+
+    @Override
+    public Map<BahmniDrugOrder, String> getMergedDrugOrderMap(List<BahmniDrugOrder> drugOrderList) {
+        Map<BahmniDrugOrder, String> mergedDrugOrderMap = new LinkedHashMap<>();
+        for(BahmniDrugOrder drugOrder : drugOrderList) {
+            BahmniDrugOrder foundDrugOrder = mergedDrugOrderMap.entrySet().stream()
+                    .map(x -> x.getKey())
+                    .filter( existingOrder ->
+                            compareDrugOrders(existingOrder, drugOrder) )
+                    .findFirst()
+                    .orElse(null);
+            if (foundDrugOrder!=null) {
+                String durationWithUnits = mergedDrugOrderMap.get(foundDrugOrder);
+                if(drugOrder.getDurationUnits() == foundDrugOrder.getDurationUnits())
+                    durationWithUnits = (foundDrugOrder.getDuration()+drugOrder.getDuration()) + " " + drugOrder.getDurationUnits();
+                else
+                    durationWithUnits += " + " + drugOrder.getDuration() + " " + drugOrder.getDurationUnits();
+                mergedDrugOrderMap.put(foundDrugOrder, durationWithUnits);
+            } else {
+                mergedDrugOrderMap.put(drugOrder, drugOrder.getDuration()+" "+drugOrder.getDurationUnits());
+            }
+        }
+        return mergedDrugOrderMap;
+    }
+
+    @Override
+    public String getPrescriptionAsString(Map<BahmniDrugOrder, String> drugOrderDurationMap, Locale locale) {
+        String prescriptionString = "";
+        int counter = 1;
+        for (Map.Entry<BahmniDrugOrder, String> entry : drugOrderDurationMap.entrySet()) {
+            prescriptionString += counter++ + ". " + getDrugOrderAsString(entry.getKey(), drugOrderDurationMap.get(entry.getKey()), locale) + "\n";
+        }
+        return prescriptionString;
+    }
+
+    public Set getUniqueProviderNames(List<BahmniDrugOrder> drugOrders) {
```

Comment:
this does not need to be a public method. Its used only SharePrescriptionServiceImpl, which identifies it from drugOrderList. This can just be in the SharePrescriptionServiceImpl implementation

Reply:
refactored to return list of unique providers and moved to SharePrescriptionServiceImpl as private method